Mathlib/Data/Prod/Basic.lean

```diff
@@ -140,6 +140,10 @@ theorem map_id : Prod.map (@id α) (@id β) = id :=
   id_prod
 #align prod.map_id Prod.map_id
 
+@[simp]
+theorem map_iterate (f : α → α) (g : β → β) (n : ℕ) :
+    (Prod.map f g)^[n] = Prod.map f^[n] g^[n] := by induction n <;> simp [*, Prod.map_comp_map]
+
```

Comment:
Please keep the #align statement: you can change the second item to account for the rename.
```suggestion
#align function.iterate_prod_map Prod.map_iterate

Reply:
In addition, mathlib policy is that whenever you rename a lemma, you add an alias for the old name with a deprecation warning. (This way, code using the old name still compiles, but emits a warning pointing to the new name --- this makes for a much nicer experience in projects depending on mathlib.) Please add such an alias; the syntax is like so:
`@[deprecated (since := "2024-07-02")] alias Function.iterate_prod_map := Prod.map_iterate`.

Reply:
Done. Thanks for the information!

Reply:
Thanks - can you move the deprecation alias to the new file as well? I *think* this is a more natural place...
and I presume the dynamics file imports `Data/Prod/Basic` anyway, so this has no consequences for anybody importing `Dynamics`.

Reply:
Done! Just have to wait for the thing to build.